docs/writing-rules/rule-syntax.md

```diff
@@ -154,6 +154,35 @@ Note that `focus-metavariable: $ARG` is not the same as `pattern: $ARG`! Using `
 
 In short, `focus-metavariable: $X` is not a pattern in itself, it does not perform any matching, it only focuses the matching on the code already bound to `$X` by other patterns. Whereas `pattern: $X` matches `$X` against your code (and in this context, `$X` only matches expressions)!
 
+
+#### Using multiple `focus-metavariable`s
+
+Use multiple focus metavariables in one Semgrep rule by specifying each under a new `focus-metavariable` item in a YAML list. With this rule, Semgrep matches all pieces of code captured by the focus metavariables.
+
+<iframe src="https://semgrep.dev/embed/editor?snippet=493E" border="0" frameBorder="0" width="100%" height="432"></iframe>
+
+There are two methods of using multiple metavariables. Each of them can lead to different results:
+
+- **Recommended**: Including `focus-metavariable` with more metavariables directly under the `patterns` key returns results regardless of whether there is an overlap between these variables:
+    ```yaml
+        patterns:
+          - pattern: foo($X, ..., $Y)
+          - focus-metavariable: 
+            - $X
+            - $Y
+    ```
+    See the following example:
+    TODO IFRAME 
+- **Advanced**: Including more `focus-metavariable` keys with different metavariables under the `pattern` key lets Semgrep find **only** the overlapping region of all the focused code:
+    ```yaml
+        patterns:
+          - pattern: foo($X, ..., $Y)
+          - focus-metavariable: $X
+          - focus-metavariable: $Y
+    ```
+    See the following example:
+    <iframe src="https://semgrep.dev/embed/editor?snippet=e8OL" border="0" frameBorder="0" width="100%" height="432"></iframe>
```

Comment:
@enncoded thank you! I think that "how" was ok in this case, but I used a different syntax here "There are two methods of using multiple metavariables."

I left "TODO IFRAME" for us to later add the Playground example. :-)

I am not sure what would be better than "Recommended" in our case - I do not think we are saying that the other example is wrong - we only say that the other method is "Advanced". Looking at [Merrian Webster's synonyms](https://www.merriam-webster.com/thesaurus/recommended), I am not sure really. "Common" does not catch what we are trying to say here ("Use mostly this method, do not be afraid to try the other also").

Reply:
Maybe we can say "Standard" instead of "Recommended" but it suggest that "Standard" is somehow lower quality than "Advanced", which is not what we want.

Reply:
Hm makes sense. Yeah we can just leave it as it is now.